File: module-contracts/src/main/java/com/beat/contracts/booking/MakerTicketReadPort.java
package com.beat.contracts.booking;

import java.util.List;

import com.beat.contracts.booking.readmodel.MakerTicketBookingStatus;
import com.beat.contracts.booking.readmodel.MakerTicketListItemReadModel;
import com.beat.contracts.booking.readmodel.MakerTicketScheduleNumber;

public interface MakerTicketReadPort {

	List<MakerTicketListItemReadModel> findTickets(Long performanceId, List<MakerTicketScheduleNumber> scheduleNumbers,
		List<MakerTicketBookingStatus> bookingStatuses);

	List<MakerTicketListItemReadModel> searchTickets(Long performanceId, String searchWord,
		List<MakerTicketScheduleNumber> scheduleNumbers, List<MakerTicketBookingStatus> bookingStatuses);
}

File: module-contracts/src/main/kotlin/com/beat/contracts/booking/MakerTicketReadPort.kt
package com.beat.contracts.booking

import com.beat.contracts.booking.readmodel.MakerTicketBookingStatus
import com.beat.contracts.booking.readmodel.MakerTicketListItemReadModel
import com.beat.contracts.booking.readmodel.MakerTicketScheduleNumber

interface MakerTicketReadPort {

    fun findTickets(
        performanceId: Long?,
        scheduleNumbers: List<MakerTicketScheduleNumber>,
        bookingStatuses: List<MakerTicketBookingStatus>,
    ): List<MakerTicketListItemReadModel>

    fun searchTickets(
        performanceId: Long?,
        searchWord: String?,
        scheduleNumbers: List<MakerTicketScheduleNumber>,
        bookingStatuses: List<MakerTicketBookingStatus>,
    ): List<MakerTicketListItemReadModel>
}
